tests/snuba/search/test_backend.py

```diff
@@ -2051,7 +2059,209 @@ def test_query(query):
+
+    def test_performance_query(self):
+        with self.feature("organizations:performance-issues"):
+            results = self.make_query(search_filter_query="issue.category:performance my_tag:1")
+        assert set(results) == {self.perf_group_1}
+
+    def test_error_performance_query(self):
+        with self.feature("organizations:performance-issues"):
+            results = self.make_query(search_filter_query="my_tag:1")
+        assert set(results) == {
+            self.perf_group_1,
+            self.error_group_1,
+            self.perf_group_2,
+            self.error_group_2,
+        }
+
+    def test_cursor_performance_issues(self):
+        # adding a couple events so there is a first and last seen
+        event1 = self.store_event(
+            data={
+                "fingerprint": ["put-me-in-group1"],
+                "event_id": "a" * 32,
+                "timestamp": iso_format(self.base_datetime - timedelta(days=21)),
+            },
+            project_id=self.project.id,
+        )
+        self.store_event(
+            data={
+                "fingerprint": ["put-me-in-group1"],
+                "event_id": "b" * 32,
+                "timestamp": iso_format(self.base_datetime),
+            },
+            project_id=self.project.id,
+        )
+        throwaway_group = Group.objects.get(id=event1.group.id)
+        throwaway_group.save()
+        self.store_group(throwaway_group)
+        perf_group_1 = self.create_group(type=GroupType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES.value)
+        perf_group_2 = self.create_group(type=GroupType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES.value)
+
+        def hack_pull_out_data(jobs, projects):
+            _pull_out_data(jobs, projects)
+            for job in jobs:
+                job["event"].groups = [perf_group_1]
+            return jobs, projects
+
+        transaction_event_data = {
+            "level": "info",
+            "message": "ayoo",
+            "type": "transaction",
+            "culprit": "app/components/events/eventEntries in map",
+            "contexts": {"trace": {"trace_id": "b" * 32, "span_id": "c" * 16, "op": ""}},
+            "tags": {
+                "hellboy": "cute",
+            },
+        }
+        events = []
+
+        for dt in [
+            throwaway_group.first_seen + timedelta(days=1),
+            throwaway_group.first_seen + timedelta(days=2),
+            throwaway_group.last_seen + timedelta(days=1),
+        ]:
+            with mock.patch("sentry.event_manager._pull_out_data", hack_pull_out_data):
+                event = self.store_event(
+                    data={
+                        **transaction_event_data,
+                        "start_timestamp": iso_format(dt),
+                        "timestamp": iso_format(dt),
+                    },
+                    project_id=self.project.id,
+                )
+                events.append(event)
+
+        for event in events:
+            event.group = perf_group_2
```

Comment:
I think we should be able to use the data from `setup` here and just paginate over it using `mytag:1` as the filter